src/test/java/com/connexta/ingest/service/impl/IngestServiceImplUnitTests.java

```diff
@@ -0,0 +1,322 @@
+
+  @Test
+  public void testIngestWhenRetrieveEndpointCreatesInvalidURI() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "this will cause the metacardLocation to be an invalid URI",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
+        .thenReturn(new URI("http://localhost:9041/mis/product/1234"));
+
+    // when
+    final StoreMetacardException storeMetacardException =
+        assertThrows(
+            StoreMetacardException.class,
+            () ->
+                ingestService.ingest(
+                    fileSize,
+                    mimeType,
+                    inputStream,
+                    fileName,
+                    metacardFileSize,
+                    metacardInputStream));
+
+    // then
+    assertThat(storeMetacardException.getReason(), equalTo("Unable to construct retrieve URI"));
+  }
+
+  /* END ingest retrieveEndpoint tests */
+
+  /* START ingest TransformClient tests */
+
+  @Test
+  public void testIngestWhenTransformClientThrowsTransformException() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    final URI location = new URI("http://localhost:9041/mis/product/1234");
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName)).thenReturn(location);
+
+    final TransformException transformException = mock(TransformException.class);
+    doThrow(transformException)
+        .when(mockTransformClient)
+        .requestTransform(eq(location), eq(mimeType), any(URI.class));
+
+    // when
+    final TransformException thrown =
+        assertThrows(
+            TransformException.class,
+            () ->
+                ingestService.ingest(
+                    fileSize,
+                    mimeType,
+                    inputStream,
+                    fileName,
+                    metacardFileSize,
+                    metacardInputStream));
+
+    // then
+    assertThat(thrown, is(transformException));
+  }
+
+  @Disabled("TODO")
+  @Test
+  public void testIngestWhenTransformClientThrowsThrowable() {}
```

Comment:
:question: Does this test still need to be written? I where Transform client is expected to throw a TransformException, but I don't see how it would be handled differently than the superclass Throwable.

Reply:
Right now our exception handlers do not handle Throwables, and that case is not unit-tested. I think the handling still needs to be implemented.